<launch>

  <!-- Utilizziamo il tempo della simulazione invece di quello dei bag file. -->
  <param name="use_sim_time" value="true"/>

  <!-- bag file -->
  <arg name="bags_path" default="/home/giacomo/catkin_ws/" />
  <arg name="bag_02" default="$(arg bags_path)Record02_0.bag $(arg bags_path)Record02_1.bag $(arg bags_path)Record02_2.bag" />
  <arg name="bag_03" default="$(arg bags_path)Record03_0.bag $(arg bags_path)Record03_1.bag $(arg bags_path)Record03_2.bag" />
  <arg name="bag" default="$(arg bag_02)" />

  <arg name="rosbag_args" default="--clock --queue=2000 --delay=0.5 --pause --hz=2000" />

  <node launch-prefix="xterm -e" pkg="rosbag" type="play" name="player" output="screen" required="true" args="$(arg rosbag_args) $(arg bag)"/>

  <arg name="depth_camera_info" default="/kinect2/qhd/camera_info"/>
  <arg name="depth_topic" default="/kinect2/qhd/image_depth_rect"/>
  <arg name="image_topic" default="/kinect2/qhd/image_color_rect"/>
  <arg name="load_model" default="false"/>
  <arg name="min_confidence" default="0.50"/>
  <arg name="model_file" default=""/>
  <arg name="output_frame" default="world"/>
  <arg name="publish_world_tf" default="true"/>
  <arg name="redirected_image_topic" default="redirected_image_gui"/>
  <arg name="z_threshold" default="0.04"/>

  <!--<include file="$(find opt_calibration)/launch/opt_calibration_results.launch" if="$(arg publish_world_tf)"/>-->

  <include file="$(find tld_tracker)/launch/tld.launch">
    <arg name="image_topic" value="$(arg image_topic)"/>
    <arg name="redirected_image_topic" value="$(arg image_topic)"/>
    <arg name="load_model" value="$(arg load_model)"/>
    <arg name="model_file" value="$(arg model_file)"/>
  </include>

  <include file="$(find bb_to_world)/launch/bb_to_world.launch">
    <arg name="depth_camera_info" value="$(arg depth_camera_info)"/>
    <arg name="depth_image" value="$(arg depth_topic)"/>
    <arg name="min_confidence" value="$(arg min_confidence)"/>
    <arg name="output_frame" value="$(arg output_frame)"/>
    <arg name="z_threshold" value="$(arg z_threshold)"/>
  </include>

  <!--Kalman fusion and `world odom` transform -->
  <include file="$(find robot_localization)/launch/ekf_sphero.launch"/>

</launch>
